### PR TITLE
[Merged by Bors] - TY-2485 add page size to configs

### PR DESCRIPTION
**Pull request by finiteprods**
previously, #147 #172

**Summary**

adds `page_size` to `EndpointConfig` for use in `Ops::new_items` impls.

### CODE REVIEW

File: discovery_engine_core/core/src/engine.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use std::{collections::HashMap, sync::Arc};
+use std::{collections::HashMap, sync::Arc, usize};
```
Comment (janpetschexain) [inline discovery_engine_core/core/src/engine.rs:15]: i guess this was accidentally added by the IDE? it's in the prelude anyways.
```suggestion
use std::{collections::HashMap, sync::Arc};
```

File: discovery_engine_core/core/src/engine.rs [hunk 2]
```diff
@@ -81,7 +81,7 @@ pub struct InitConfig {
     pub api_key: String,
     /// API base url.
     pub api_base_url: String,
-    /// List of markets to use.
+    /// List of markets to use
```
Comment (janpetschexain) [inline discovery_engine_core/core/src/engine.rs:84]: is this reverted on purpose? all other field docs had a period added recently.

**Review by janpetschexain — APPROVED** [review]
lgtm

File: discovery_engine_core/core/src/engine.rs (hunk 1 above)
Comment (finiteprods) [reply discovery_engine_core/core/src/engine.rs]: i guess it was added accidentally, good catch.

File: discovery_engine_core/core/src/engine.rs (hunk 2 above)
Comment (finiteprods) [reply discovery_engine_core/core/src/engine.rs]: perhaps a mix up while fixing merge conflicts - thanks.

**Comment by finiteprods** [comment]
bors merge

**Pull request closed without merging**